include/pisa/query/parser.hpp

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <functional>
+#include <optional>
+#include <string>
+
+#include "query.hpp"
+
+namespace pisa {
+
+using TermResolver = std::function<std::optional<ResolvedTerm>(std::string)>;
+
+struct StandardTermResolverParams;
+
+/// Provides a standard implementation of `TermResolver`.
+class StandardTermResolver {
```

Comment:
Yes, I want this to replace the other but in order to avoid a large PR, I introduce this first, before substituting it for all tools, when in turn we'll be able to remove it.